xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.h

```diff
@@ -38,12 +50,74 @@ class CDVDSubtitlesLibass
 
   bool DecodeHeader(char* data, int size);
   bool DecodeDemuxPkt(const char* data, int size, double start, double duration);
+
+  /*!
+  * \brief Create a new ASS track with a default ASS style (used on ASS events)
+  * \param useDebugStyle Use style text/color/position for debug info purpose
+  * \return True if success, false if error
+  */
+  bool CreateTrack(bool useDebugStyle = false);
+  /*!
+  * \brief Create a new ASS track based on an SSA buffer
+  * \return True if success, false if error
+  */
   bool CreateTrack(char* buf, size_t size);
 
+  /*!
+  * \brief Add an ASS event to show a subtitle on a specified time
+  * \param startTime The PTS start time of the Event
+  * \param stopTime The PTS stop time of the Event
+  * \return Return the Event ID, otherwise ASSEVENT_NO_ID if fails
+  */
+  int AddEvent(std::string text, double startTime, double stopTime);
+
+  /*!
+  * \brief Append text to the specified event (\\n will be added before the text)
+  */
+  void AppendTextToEvent(int eventId, std::string text);
```

Comment:
any methods that are not consumed by any ASS/SSA class (ass parser and ass codec) shouldn't be publicly exposed to anyone else. Maybe your `KodiSubtitleAdapter` (my first comment) can be a friend class of this one

Reply:
done